## Results {.page_break_before}

ERDs are repaired more efficiently than other replication domains. 
To determine how repair preferences influenced by replication domains, 
we intersected our Damage-seq and XR-seq data with localized replication domains of HeLa cells 
and plotted the normalized repair rates of replication domains for early and late phases. 
We find that ERDs are repaired higher than the other replication domains 
even though it decreases at late phase. 
On the contrary, LRDs and transition zones (UTZs, DTZs) 
which have lower repair rates 
increased at late phase (Fig 1a). 

LRDs and transition zones repaired more dominantly at late phase, 
while the repair of ERDs are dominant at early phase. 
To display the replication timing effect on replication domains more clearly, 
we plotted early/late phase values of replication domains. 
As expected, except ERDs, all the other replication domains (LRDs, UTZs, DTZs) repaired better at late phase (Fig 1b). 
Then we separated replication domains into 10 kb long bins covering total of 2 Mbp (200 bins) 
where replication domains are centered. 
We observe that repair rates of ERDs decrease 
when the replication go through up and downstreams of ERDs. 
Also, ERDs are repaired more efficiently at early phase than it is late phase. 
Furthermore, at the center, the repair rate difference between early and late phase is the highest, 
and repair rates of phases gets closer at the sides (Fig 1c). 
Conversely, LRDs’ repair efficiency is very low at the center, 
increases trough up and downstreams. 
Like ERDs, LRDs’ repair rate difference between phases are at the utmost 
and gets lower at the sites. 
However, LRDs’ repair efficiency is higher at late phases (Fig 1d). 
Moreover, repair efficiency of UTZs and DTZs are similar to LRDs (Supp Fig 1). 
In general, (6-4)PP damages display less difference in contrast to CPD damages. 
The reason behind this fenomena is 
that (6-4)PPs repaired distinctly faster and earlier than CPDs. 
Until the 12 minutes past 
which is the duration between UV exposure and sequencing, 
most of the (6-4)PP damages are repaired.

Replication domains are repaired differentially between chromosomes. 
Next, we asked whether there is any difference at repair preferences of chromosomes. 
Since (6-4)PPs do not display much difference, 
we continued with CPD damages only. 
Interestingly, we find some chromosomes repaired much higher rates at both ERDs and LRDs. 
Especially chromosome 21 is repaired considerably better than all other chromosomes (Fig 2a). 
Conversely, chromosome X’s repair rates are one of the lowest, 
might be caused by X-inactivation 
which turns the chromatin state of one of the X chromosomes into heterochromatin structure.  

Chromatin states effect the repair efficiency. 
Since NER can work more efficiently at the open chromatin sites, 
we decided to analyze repair rates of chromatin states. 
We used 22 chromatin states that generated by ChromHMM. 
Notably, both ERDs and LRDs are repaired more efficiently at open chromatin states. 
At the active promoters (Tss, TssF) and candidate strong enhancers (Enh, EnhF), 
repair rates of LRDs almost equal to ERDs. 
Even though we observe higher ERD repair rate in most of the open chromatin states, 
some reveal a different profile. 
Surprisingly, a weak enhancer called FaireW, 
which contains the regulatory element locations discovered by Faire-seq, 
has a repair rate as low as repressed chromatin states at both ERD and LRD domains. 
Moreover, chromatin states with condensed regions have lower repair rates, 
and the interval between repair rates of ERDs and LRDs increases (Fig 3a).  

We also want to see how repair preferences at chromatin states change through the phases. 
Generally, from early to late phase, 
repair rate differences get decreased by ERDs getting repaired less efficiently and 
LRDs getting repaired better. 
However, at some repressed regions, such as Repr chromatin state, 
even though the interval between repair rates get lower, 
the repair rate of ERD at late phase (sharp, blue framed point) is higher than the repair rate at early phase (shady, blue framed point). 
This phenomenon might have occurred because 
when replication moves on, 
even the heterochromatin regions will be open at later phases and 
become more reachable for NER and 
ERDs at condensed regions are repaired better 
when the chromatin opens (Fig 3a). 
Then we asked if ERDs’ repair rate decrease, 
or LRDs’ repair rate increase is higher 
when replication timing moves from early to late phase. 
To answer our question, 
we plotted early/late values of ERDs and late/early values of LRD to the same graph. 
If LRD is the one with the higher value, 
it means its increase is higher than the decrease of ERD 
and if ERD is higher, vice versa. 
Except FaireW, at the enhancer chromatin states (orange and yellow), 
the decrease of ERDs higher than the increase of LRDs. 
At all the other chromatin states, LRDs’ increase beats the ERDs’ decrease. 
Also, as we mentioned before with Repr chromatin state, 
even though we expect ERDs to get decreased at the late phases, 
for all repressed chromatin states ERDs are repaired better at late phase. 
Conversely, LRDs at enhancers - except FaireW - are repaired more efficiently at early phase,
probably because those LRDs are not as condensed as repressed regions (Fig 3b). 

![Experimental Setup. Separate HeLa cells are UV irradiated at different stages of cell cycle: early S phase, late S phase, and asyncronized S phase. While half of the early and late phased samples are collected after 12 minutes, the other half are left for 2 hours. XR-seq (XR) and Damage-seq (DS) methods are applied to samples that are collected after 12 minutes for both CPD and (6-4)PP damages. However, for samples that are collected after 2 hours, the methods are applied only to CPD damage, due to the fast repair ability of the (6-4)PP damages which will already be mostly repaired after 2 hours. Asyncronized cells are collected only after 12 minutes. DNA-seq method is applied at all 3 phases. Except DNA-seq, all the methods are applied twice.](images/){[2019.12.10]experimental_setup}


| *Cell Line* | Product | Method | Phase | Time after Exposure (min) | Replicate |
|:--------------:|:----------:|:----------:|:----------:|:----------:|:--------------|
HeLa-S3 | (6-4)PP | XR-seq | Early | 12 | A | B |
HeLa-S3 | (6-4)PP | XR-seq | Late | 12 | A | B |
HeLa-S3 | (6-4)PP | XR-seq | Async | 12 | A | B |
HeLa-S3 | (6-4)PP | Damage-seq | Early | 12 | A | B |
HeLa-S3 | (6-4)PP | Damage-seq | Late | 12 | A | B |
HeLa-S3 | (6-4)PP | Damage-seq | Async | 12 | A | B |
HeLa-S3 | CPD | XR-seq | Early | 12 | A | B |
HeLa-S3 | CPD | XR-seq | Late | 12 | A | B |
HeLa-S3 | CPD | XR-seq | Async | 12 | A | B |
HeLa-S3 | CPD | Damage-seq | Early | 12 | A | B |
HeLa-S3 | CPD | Damage-seq | Late | 12 | A | B |
HeLa-S3 | CPD | Damage-seq | Async | 12 | A | B |
HeLa-S3 | CPD | XR-seq | Early | 120 | A | B |
HeLa-S3 | CPD | XR-seq | Late | 120 | A | B |
HeLa-S3 | CPD | Damage-seq | Early | 120 | A | B |
HeLa-S3 | CPD | Damage-seq | Late | 120 | A | B |
HeLa-S3 | NA | DNA-seq | Early | NA | A |
HeLa-S3 | NA | DNA-seq | Late | NA | A |
HeLa-S3 | NA | DNA-seq | Async | NA | A |

Table: Sample information.
{#tbl:samples}

| *Name* | File Format | Method | Cell Line | Genome | 
|:--------------:|:----------:|:----------:|:----------:|:----------:|
S1 to S2 regions | bigWig (6 files) | Repli-seq | HeLa-S3 | hg19 |
Replication domains | bed | Repli-seq | HeLa-S3 | hg19 | 
High RFD | bed | OK-seq | HeLa-S3 | hg19 | 
Initiation zones | bed | OK-seq | HeLa-S3 | hg19 | 
SNS-seq | bed (2 files) | SNS-seq | HeLa-S3 | hg19 | 
ChromHMM chromatin states | bed | Multiple methods | HeLa-S3 | hg19 | 
C to T mutations | json | Multiple methods | Skin | hg38 | 

Table: Data retrieved from literature.
{#tbl:retrieved}

![Length distribution of excised oligomers from CPD of HeLa cells at early phase after adaptor trimming and dublicate removal.](images/){HXECB11_GGCTAC_cutadapt_length_distribution}

![Dinucleotide frequency of 26-mers from CPD of HeLa cells at early phase.](images/){HXECB11_GGCTAC_cutadapt_sorted_26_dinucleotideTable}

![Damage distribution of the chromosomes for every sample. The damage values are RPKM normalized.](images/){[2019.11.25]RPKM_Damage_Values_of_all_Samples_at_Every_Chromosome}

![For every sample, the RPKM values of transcribed strand is divided by non-transcribed strand (TS/NTS) and log2 normalized. Above the red horizontal dashed line demonstrates that TS values are higher than NTS values, below demonstrates that NTS is higher.](images/){TS_NTS_all_log2norm}

## Replication Domains

![RPKM values of 64PP XRseq samples are divided by 64PP Damage-seq samples (Repair Rate) for both early (violet) and late (red) replication domains and log2 normalized. Above the red horizontal dashed line demonstrates that repair is higher than damage, below demonstrates that damage is higher.](images/){[2019.11.14]Repair_over_Damage_Value_of_ERD_LRD_for_64_12}

![RPKM values of CPD XRseq samples which are harvested after 12 minutes after UV exposure, are divided by equivalent CPD Damage-seq samples (Repair Rate) for both early (violet) and late (red) replication domains and log2 normalized. Above the red horizontal dashed line demonstrates that repair is higher than damage, below demonstrates that damage is higher.](images/){[2019.11.14]Repair_over_Damage_Value_of_ERD_LRD_for_CPD_12}

![RPKM values of CPD XRseq samples which are harvested after 2 hours after UV exposure, are divided by equivalent CPD Damage-seq samples (Repair Rate) for both early (violet) and late (red) replication domains and log2 normalized. Above the red horizontal dashed line demonstrates that repair is higher than damage, below demonstrates that damage is higher.](images/){[2019.11.14]Repair_over_Damage_Value_of_ERD_LRD_for_CPD_120}

![RPKM values of 64PP XRseq samples are divided by 64PP Damage-seq samples (Repair Rate) for both early (red) and late (turquoise) replication domains across the chromosomes and log2 normalized. Above the green horizontal dashed line means that repair doubles the damage, below the violet horizontal line means that damage doubles the repair.](images/){[2019.11.14]Repair_over_Damage_Value_of_ERD_LRD_for_Every_Chromosome_64_12}

![RPKM values of CPD XRseq samples which are harvested after 12 minutes after UV exposure, are divided by equivalent CPD Damage-seq samples (Repair Rate) for both early (red) and late (turquoise) replication domains across the chromosomes and log2 normalized. Above the green horizontal dashed line means that repair doubles the damage, below the violet horizontal line means that damage doubles the repair.](images/){[2019.11.14]Repair_over_Damage_Value_of_ERD_LRD_for_Every_Chromosome_CPD_12}

![RPKM values of CPD XRseq samples which are harvested after 2 hours after UV exposure, are divided by equivalent CPD Damage-seq samples (Repair Rate) for both early (red) and late (turquoise) replication domains across the chromosomes and log2 normalized. Above the green horizontal dashed line means that repair doubles the damage, below the violet horizontal line means that damage doubles the repair.](images/){[2019.11.14]Repair_over_Damage_Value_of_ERD_LRD_for_Every_Chromosome_CPD_120}

![Repair rates (XRseq/Damage-seq) are calculated and log2 normalized in 2 Mbp sliding windows with 10 kb intervals, which early replication domains (ERDs) positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.15]Repair_over_Damage_Value_of_ERD_for_Every_Sample}

![Repair rates (XRseq/Damage-seq) are calculated and log2 normalized in 2 Mbp sliding windows with 10 kb intervals, which late replication domains (LRDs) positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.15]Repair_over_Damage_Value_of_LRD_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also early phase samples are divided by late phase samples (early/late) in 2 Mbp sliding windows with 10 kb intervals, which early replication domains (ERDs) positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.15]Early_over_Late_Phase_Value_of_Repair_over_Damage_Value_of_ERD_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also early phase samples are divided by late phase samples (early/late) in 2 Mbp sliding windows with 10 kb intervals, which late replication domains (LRDs) positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.15]Early_over_Late_Phase_Value_of_Repair_over_Damage_Value_of_LRD_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also minus strands are divided by plus strands (minus/plus) in 2 Mbp sliding windows with 10 kb intervals, which early replication domains (ERDs) positioned at the center of the region. Red lines are the early phase samples and blue lines are the late phase samples.](images/){[2019.11.15]Minus_over_Plus_Strand_Value_of_Repair_over_Damage_Value_of_ERD_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also minus strands are divided by plus strands (minus/plus) in 2 Mbp sliding windows with 10 kb intervals, which late replication domains (LRDs) positioned at the center of the region. Red lines are the early phase samples and blue lines are the late phase samples.](images/){[2019.11.15]Minus_over_Plus_Strand_Value_of_Repair_over_Damage_Value_of_LRD_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also minus strands are divided by plus strands (minus/plus) in 2 Mbp sliding windows with 10 kb intervals, which early and late replication domains are positioned at the center of the region. Red lines are the early phase samples and blue lines are the late phase samples.](images/){[2019.11.15]Minus_over_Plus_Strand_Value_of_Repair_over_Damage_Value_of_ERD_LRD_for_Every_Sample}

## High RFDs

![Repair rates (XRseq/Damage-seq) are calculated and log2 normalized in 200 kb sliding windows with 20 kb intervals, which regions with high replication fork directinality (high RFD) positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.17]Repair_over_Damage_Value_of_HiRFDs_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also early phase samples are divided by late phase samples (early/late) in 200 kb sliding windows with 20 kb intervals, which regions with high replication fork directinality (high RFD) positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.17]Early_over_Late_Phase_Value_of_Repair_over_Damage_Value_of_HiRFDs_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also minus strands are divided by plus strands (minus/plus) in 200 kb sliding windows with 20 kb intervals, which regions with high replication fork directinality (high RFD) are positioned at the center of the region. Red lines are the early phase samples and blue lines are the late phase samples.](images/){[2019.11.17]Minus_over_Plus_Strand_Value_of_Repair_over_Damage_Value_of_HiRFDs_for_Every_Sample.pdf}

## Initiation Zones

![Repair rates (XRseq/Damage-seq) are calculated and log2 normalized in 20 kb sliding windows with 100 base pair intervals, which Initiation Zones are positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.17]Repair_over_Damage_Value_of_InZones_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also early phase samples are divided by late phase samples (early/late) in 20 kb sliding windows with 100 base pair intervals, which Initiation Zones are positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.17]Early_over_Late_Phase_Value_of_Repair_over_Damage_Value_of_InZones_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also minus strands are divided by plus strands (minus/plus) in 20 kb sliding windows with 100 base pair intervals, which Initiation Zones are positioned at the center of the region. Red lines are the early phase samples and blue lines are the late phase samples.](images/){[2019.11.17]Minus_over_Plus_Strand_Value_of_Repair_over_Damage_Value_of_InZones_for_Every_Sample}

## SNS-seq

![Repair rates (XRseq/Damage-seq) are calculated and log2 normalized in 20 kb sliding windows with 100 base pair intervals, which short nascent strands (SNS) are positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.17]Repair_over_Damage_Value_of_SNSseq_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also early phase samples are divided by late phase samples (early/late) in 20 kb sliding windows with 100 base pair intervals, which short nascent strands (SNS) are positioned at the center of the region. The blue lines are the plus strands and red lines are the minus strands.](images/){[2019.11.17]Early_over_Late_Phase_Value_of_Repair_over_Damage_Value_of_SNSseq_for_Every_Sample}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also minus strands are divided by plus strands (minus/plus) in 20 kb sliding windows with 100 base pair intervals, which short nascent strands (SNS) are positioned at the center of the region. Red lines are the early phase samples and blue lines are the late phase samples.](images/){[2019.11.17]Minus_over_Plus_Strand_Value_of_Repair_over_Damage_Value_of_SNSseq_for_Every_Sample}

## S1 to S2 Regions

![Repair rates (XRseq/Damage-seq) are calculated and log2 normalized in 100 kb sliding windows with 1 kb intervals, which S1 to S2 regions are positioned at the center of the region. The blue lines are the leading strands and red lines are the lagging strands.](images/){[2019.10.12]Repair_over_Damage_Value_of_S1toS2_exact_peak_for_Every_Sample.pdf}

![After log2 normalized repair rates (XRseq/Damage-seq) are calculated, also early phase samples are divided by late phase samples (early/late) in 100 kb sliding windows with 1 kb intervals, which S1 to S2 regions are positioned at the center of the region. The blue lines are the leading strands and red lines are the lagging strands.](images/){[2019.10.12]Early_over_Late_Phase_Value_of_Repair_over_Damage_Value_of_S1toS2_exact_peak_for_Every_Sample.pdf}

![Histogram of calculated WA scores.](images/){[2019.11.14]histogram_WA}

## Mutation Profile of Initiation Zones

![C to T mutations are mapped to Initiation Zones and counted in 20 kb sliding windows with 100 base pair intervals. Red lines are the plus strands and blue lines are the minus strands.](images/){[2019.11.14]Mutation_Profile_of_InitiationZones_200_sliding_windows_at_100bp_intervals}