\setlength{\parindent}{0pt}
\chapter{\bf The Scope of the Thesis}

Nucleotide excision repair is the sole mechanism for the removal of bulky adducts. In this study, to assess the influence of replication along with nucleotide excision repair on mutation distribution across replicated sites, we analyzed the Damage-seq and eXcision Repair sequencing (XR-seq) data. Damage and repair maps are generated for cyclobutane pyrimidine dimers (CPDs) and pyrimidine-pyrimidone (6-4) photoproducts [(6-4)PPs] from UV-irradiated HeLa cells synchronized at two stages of the cell cycle: early S phase, and late S phase. Damage-seq locates and quantifies the regions of UV induced CPD and (6-4)PP damages, while XR-seq captures excised oligomers of the damage site that are removed by the nucleotide excision repair. Two methods combined provide the genome-wide distribution of UV-induced damages and the differential repair frequency of these damage sites. 

Initially, we examined the quality of the reads that are produced by Damage-seq and XR-seq methods. After quality filtering and performing pre-analysis of damage and repair reads, we prepared bed files containing the positions of damage and repair events on the human genome. Then, we used these bed files together with data sets obtained from public sources to compare the repair rate of nucleotide excision repair at different regions.

In the first part of the study, we mapped the damage and repair events to the replication domains, where closely packed origin of replications fire in a synchronized manner, resulting in simultaneous replication of these Mb-sized regions. Then, we normalized repair events with corresponding damage quantities to eliminate the potential bias caused by the damage formation. By doing so, we managed to observe the differential repair rate between replication domains at different time points on a wide scale. We performed a similar analysis using chromatin states of HeLa cells and examined how chromatin states effect the repair rate of replication domains, while moving early to late S phase of cell cycle.

Secondly, we aim to find whether nucleotide exicision repair contribute to a replicative strand asymmetry. Because nucleotide excision repair is highly associated with melanoma cancers, replicative strand asymmetry of nucleotide excision repair can correlate with the mutation profiles of melanoma cancers. We retrieved a somatic melanoma mutations data, and quantified the mutations on approximately 20 kb-sized initiation zones where origin of replications closely positioned. We further separated these initiation zones into their corresponding replication domains before quantifying the mutations. This method enabled us both to compare the mutation count differences of replication domains, and to observe the mutational strand asymmetry on initiation zones. Next, we examined the strand asymmetry of damage and repair events separately on initiation zones. To see if nucleotide composition of initiation zones is contributing to the strand asymmetry, we simulated Damage-seq and XR-seq reads, and compared the signal levels of these reads on initiation zones as well. Lastly, we calculated the repair rate by normalizing repair events with damage quantities and mapped them to observe the asymmetry of repair rate.